controller/src/beerocks/master/db/network_map.cpp

```diff
@@ -109,20 +115,41 @@ void network_map::send_bml_network_map_message(db &database, Socket *sd,
                 response =
                     message_com::create_vs_message<beerocks_message::cACTION_BML_NW_MAP_RESPONSE>(
                         cmdu_tx, id);
-                beerocks_header                      = message_com::get_beerocks_header(cmdu_tx);
-                beerocks_header->actionhdr()->last() = 0;
-                num_of_nodes = response->node_num(); // prepare for next message
-                num_of_nodes = 0;
-                data_start   = nullptr;
-                size         = 0;
+                if (!response) {
+                    LOG(ERROR) << "create_vs_message() failed ";
+                    return;
+                }
+
+                beerocks_header = message_com::get_beerocks_header(cmdu_tx);
+                if (!beerocks_header) {
+                    LOG(ERROR) << "get_beerocks_header() failed ";
+                    return;
+                }
+
+                auto actionhdr = beerocks_header->actionhdr();
+                if (!actionhdr) {
+                    LOG(ERROR) << "actionhdr() failed ";
+                    return;
+                }
+
+                actionhdr->last() = 0;
+                num_of_nodes      = response->node_num(); // prepare for next message
+                num_of_nodes      = 0;
```

Comment:
I realize that you didn't change this code, but something is fishy here - the first assignment is redundant...

Reply:
Yes, I saw that too, @morantr can I remove line 136?

Reply:
For something like this you don't need approval from Moran.. the first assignment is simply redundant so just remove it

Reply:
Actually it is a serious bug now that I look at it. `num_of_nodes` is defined as reference of `response->node_num()` at the function beginning. On that block, we create a new message but we still use the reference to the previous message. It is accidentally working since the `num_of_nodes` location on the buffer is not changed when creating a new message.

I think that regardless of deleting the redundant line, we need also to use `response->node_num()` directly instead of the reference, and delete the reference declaration at the function beginning.